### PR TITLE
Simplify fitting

### PR DESCRIPTION
**Pull request by Jacob-Stevens-Haas**

### CODE REVIEW

File: pysindy/pysindy.py [hunk 1]
```diff
@@ -665,13 +630,12 @@ def score(
...
-        if not isinstance(self.feature_library, WeakPDELibrary):
-            x, x_dot = zip(
-                *[drop_nan_samples(xi, xdoti) for xi, xdoti in zip(x, x_dot)]
-            )
+        x_dot, x_dot_predict = zip(
+            *[
+                drop_nan_samples(xdoti, xdotpi)
+                for xdoti, xdotpi in zip(x_dot, x_dot_predict)
+            ]
+        )
```
Comment (Jacob-Stevens-Haas) [inline pysindy/pysindy.py:638]: Do we need to drop from x, x_dot, and x_dot_predict?

File: pysindy/optimizers/base.py [hunk 2]
```diff
@@ -193,16 +191,6 @@ def fit(self, x_, y, sample_weight=None, **reduce_kws):
...
-class _MultiTargetLinearRegressor(MultiOutputRegressor, ComplexityMixin):
-    @property
-    def coef_(self):
-        return np.vstack([est.coef_ for est in self.estimators_])
-
-    @property
-    def intercept_(self):
-        return np.array([est.intercept_ for est in self.estimators_])
-
-
```
Comment (Jacob-Stevens-Haas) [inline pysindy/optimizers/base.py:205]: Why remove?

File: pysindy/pysindy.py (hunk 1 above)
Comment (znicolaou) [reply pysindy/pysindy.py:638]: If the user defines the differentiation method to drop endpoints, x_dot will be nan there, which should cause problems unless we drop them. I don't think there's a test for score with differentiation_method, but I'll write one.

File: pysindy/optimizers/base.py (hunk 2 above)
Comment (znicolaou) [reply pysindy/optimizers/base.py:205]: I didn't think this was being invoked ever. Do any optimizers have multiple estimators?

Comment (Jacob-Stevens-Haas) [reply pysindy/optimizers/base.py:205]: I assumed this was for fitting vector ODEs, i.e. the Lorentz equations.  SINDyOptimizer previously wrapped it's underlying estimator in `_MultiTargetLinearRegressor()` when `y.shape[1] > 1`.  Deleting it doesn't seem to cause it to fail tests, though, so :man_shrugging:

**Pull request merged**